src/Stratis.Bitcoin.IntegrationTests.Common/EnvironmentMockUpHelpers/NodeBuilder.cs

```diff
@@ -183,5 +184,58 @@ public void Dispose()
             LogManager.Configuration.LoggingRules.Clear();
             LogManager.ReconfigExistingLoggers();
         }
+
+        /// <summary>
+        /// By default, logs are disabled when using Create() method, so using this fluent method, the caller can enable the logs at will.
+        /// </summary>
+        /// <example>
+        /// //default use (without logs)
+        /// using (NodeBuilder builder = NodeBuilder.Create(this))
+        /// {
+        ///     //your test code here
+        /// }
+        ///
+        /// //with logs enabled
+        /// using (NodeBuilder builder = NodeBuilder.Create(this).WithLogsEnabled())
+        /// {
+        ///     //your test code here
+        /// }
+        /// </example>
+        /// <returns>Current NodeBuilder instance, used for fluent API style</returns>
+        public NodeBuilder WithLogsEnabled()
+        {
+            if (!LogManager.IsLoggingEnabled())
+            {
+                // NLog Enable/Disable logging is based on internal counter, so to be sure to have enabled logs, must to
+                // keep calling EnableLogging until IsLoggingEnabled returns true.
+                do
+                {
+                    LogManager.EnableLogging();
+                }
+                while (!LogManager.IsLoggingEnabled());
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// If logs have been enabled by calling WithLogsEnabled you can disable it manually by colling this method.
+        /// If the test is running within an "using block" where the nodebuilder is created, without using WithLogsEnabled,
+        /// you shouldn't need to call this method.
+        /// </summary>
+        /// <returns>Current NodeBuilder instance, used for fluent API style</returns>
+        public NodeBuilder WithLogsDisabled()
+        {
+            if (LogManager.IsLoggingEnabled())
+            {
+                // NLog Enable/Disable logging is based on internal counter, so to be sure to have enabled logs, must to
+                // keep calling DisableLogging until IsLoggingEnabled returns false.
+                do
+                {
+                    LogManager.DisableLogging();
+                }
+                while (LogManager.IsLoggingEnabled());
+            }
+            return this;
```

Comment:
also here

Reply:
i did some changes, how do you want the `while` in a do/while block? I didn't found it on code style, should the while be on the same line of the closing bracket? if on a new line, should i always leave a blank line between the while and the next closing bracket?